addons/purchase_stock/static/tests/tours/purchase_catalog_suggest.js

```diff
@@ -37,106 +25,150 @@ registry.category("web_tour.tours").add("test_purchase_order_suggest_search_pane
+        ...productCatalog.addProduct("test_product"),
+        ...productCatalog.waitForQuantity("test_product", 24),
+
+        //  ---- UI should hide suggestion if in the order qty = suggested_qty
+        { trigger: "div[name='kanban_purchase_suggest'] span:hidden" }, // If qty in PO == suggested_qty --> hide suggest string
+        ...productCatalog.addProduct("test_product"),
+        ...productCatalog.waitForQuantity("test_product", 25),
+        { trigger: "div[name='kanban_purchase_suggest'] span:visible" }, // If qty in PO != suggested_qty --> show suggest string
+
+        // -- The quantity in the catalog and order line should be equal
+        ...productCatalog.goBackToOrder(),
+        ...purchaseForm.checkLineValues(0, { product: "test_product", quantity: "25.00" }),
+        ...purchaseForm.openCatalog(),
+        ...productCatalog.removeProduct("test_product"),
+        // Should go back to displaying suggested qtys
+        ...catalogSuggestion.assertCatalogRecord("test_product", { monthly: 52, suggest: 24 }),
+        ...catalogSuggestion.checkKanbanRecordPosition("test_product", 0),
+        /*
+         * -------------------  PART 4 : KANBAN FILTERS ---------------------
+         * Checks suggest and searchModel (filters) interactions
+         * (Add / Remove with filters), category filters
+         * ------------------------------------------------------------------
+         */
+
+        // ---- Check Adding non suggested product works with suggest
+        ...catalogSuggestion.toggleSuggest(false),
+        ...productCatalog.addProduct("Courage"),
+        ...productCatalog.waitForQuantity("Courage", 1),
+        ...catalogSuggestion.toggleSuggest(true),
+
+        // ---- Check toggling suggest OFF with filters manually removed still works
+        ...catalogSuggestion.removeSuggestFilter(),
+        ...catalogSuggestion.toggleSuggest(false),
+        ...catalogSuggestion.checkKanbanRecordPosition("Courage", 0), // == suggest is off
+
+        // --- Turning suggest on with non suggested product works as expected
+        // Because Add product can be slow to reach server and because when toggling suggest we filter
+        // products in the order, we go back to catalog and come back (similar to an await).
+        // (This will probably not be need in following PR when we remove debounce on AddProduct)
+        ...productCatalog.goBackToOrder(),
+        ...purchaseForm.openCatalog(),
+        ...catalogSuggestion.toggleSuggest(true),
+        ...catalogSuggestion.checkKanbanRecordPosition("Courage", 1), // Courage still shown because in order but after suggested products
+
+        // Check that categories work well with suggestions
+        ...productCatalog.selectSearchPanelCategory("Goods"),
+        { trigger: "span[name='suggest_total']:visible:contains('$ 0.00')" }, // Should recompute estimated price
+        ...productCatalog.selectSearchPanelCategory("Expenses"),
```

Comment:
I don't know what changed in the meantime but the tour breaks at this step because the expenses category is not displayed (I checked, there is no existing products using this category in this tour's test environment.)
To fix that, I assigned the test category to the test product and use this category instead of the expense one.